storage: update import path for Raft

tbg opened the pull request:
upstream moved from github.com/etcd to go.etcd.io/etcd. We're going
to have to pick up an update soon to fix #28918, so it's easier to
switch now.

We're not picking up any new commits except for the renames.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/29574)
<!-- Reviewable:end -->

tbg committed:
storage: update import path for Raft

upstream moved from github.com/etcd to go.etcd.io/etcd. We're going
to have to pick up an update soon to fix #28918, so it's easier to
switch now.

We're not picking up any new commits except for the renames.

Release note: None

petermattis reviewed (dismissed):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> for the renames. Small question about the different SHA and whether a version bump now is with the risk vs your other PR which works around the problem solely in cockroach code.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/29574#-:-LLe9YZPERUZAbYEflnr:b-yofbo6)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale)

___
*[Gopkg.lock, line 1317 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/29574#-LLe9IMA1LbeODAW9hHC:-LLe9IMA1LbeODAW9hHD:b-2dqfg1) ([raw file](https://github.com/cockroachdb/cockroach/blob/3afbd217567aa86c847e9104bd928b871c295e12/Gopkg.lock#L1317)):*
> ```TOML
>   ]
>   pruneopts = "UT"
>   revision = "1df1ddff4361ed7f2c0f33571923511889a115ce"
> ```

Should I worry that this is a different sha? Are we getting an `etcd/raft` bump at the same time?


<!-- Sent from Reviewable.io -->

tbg reviewed (commented):
TFTR. There should be zero risk in this PR since we're not picking up any new code.
I'd also like to pick up the upstream fix when it's merged, though perhaps early in the 2.2 cycle.
In any case, the next time we bump Raft we'll have a good reason to do so, so it will be nice to not shave this yak then.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/29574#-:-LLeEXo65G8shiaOFGIH:bqjsjhl)* status: :shipit: complete! 1 of 0 LGTMs obtained

___
*[Gopkg.lock, line 1317 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/29574#-LLe9IMA1LbeODAW9hHC:-LLeEKJXBJTzkfbQ_yNW:by10hn7) ([raw file](https://github.com/cockroachdb/cockroach/blob/3afbd217567aa86c847e9104bd928b871c295e12/Gopkg.lock#L1317)):*
No, see

> We're not picking up any new commits except for the renames.

We are picking up the renames, so the SHA is different. See https://github.com/etcd-io/etcd/commits/master/raft


<!-- Sent from Reviewable.io -->

tbg reviewed (commented):
bors r=petermattis

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/29574#-:-LLeIsjc2ZfTuf_NLwj8:blkd80i)* status: :shipit: complete! 1 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->

tbg reviewed (commented):
> Dismissed @petermattis from a discussion.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/29574)* status: :shipit: complete! 1 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->

tbg reviewed (commented):
bors r=petermattis

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/29574#-:-LLeJ6_qEPkam_TevE3w:blkd80i)* status: :shipit: complete! 1 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->

tschottdorf commented:
bors r=petermattis

benesch commented on Gopkg.lock:
```diff
@@ -1270,14 +1259,6 @@
-  packages = ["."]
-  pruneopts = "UT"
-  revision = "bb4de0191aa41b5507caa14b0650cdbddcd9280b"
-
```
Um. This should not have gotten deleted as far as I can tell. Looking into it.

tschottdorf replied:
Thanks and apologies.

benesch replied:
Not your fault at all! That frustration was directed at dep.

tschottdorf replied:
Well, I could've seen something unexpected was happening. :-)
